### DISCUSSION TITLE
Coverage failing because of trying to read map file

### DISCUSSION THREAD

**Discussion by jasonkuhrt** [vitest-dev/vitest#4754] [Q&A]
I'm very confused by this Vitest with coverage behaviour. I am in a simple Remix app. I am just testing a trivial 1=1 code thing for now. Coverage continues to fail like so:

![CleanShot 2023-12-14 at 11 55 29@2x](https://github.com/vitest-dev/vitest/assets/284476/1ebf43b7-729f-44c7-97c4-8f50b5ed1ee0)

1. Why is it trying to read something in `/build/`?
2. Why is it trying to read a Remix file (or at least, a file with remix in the name)

Is there some magic going on given that remix is setup in the project?

**Comment by jasonkuhrt** [comment]
I deleted my `node_modules` and then reinstalled everything. Now the results are different (still failing):

![CleanShot 2023-12-14 at 12 02 31@2x](https://github.com/vitest-dev/vitest/assets/284476/60a5dc17-8a59-4f32-b9e1-f42dd137b22a)

**Comment by jasonkuhrt** [comment]
I am now trying Istanbul reporter, which yields a different error still:

![CleanShot 2023-12-14 at 12 06 46@2x](https://github.com/vitest-dev/vitest/assets/284476/380c489b-0ce4-4c05-898e-4b2635492451)

This time quite different, as if the Remix route file names are confusing it, not sure, though.

**Comment by jasonkuhrt** [comment]
It turns out I had some `./public/build/...` files. deleting those made these errors go away. Still confused, though.

1. Why didn't `public` show up in the reported paths in error messages
2. Why isn't `exclude` option of coverage working? E.g. I have to delete these files to avoid the above failures, there must be another way?

**Reply by ldevoco** [reply]
I had this bug too, if someone else has this bug, try this workaround:

I had a min.js file which had a comment in it like this:
//# sourceMappingURL=some.file.min.js.map

^ That was what triggered the bug, I removed it from the min.js file and vitest coverage stopped crashing. What I find truly a bug is that this file was explicitly excluded from my vitest config -> test -> exclude section, it shouldn't have been reading it at all.

**Reply by sheremet-va** [reply]
`test.exclude` doesn't exclude files from reading, Vitest just doesn't consider it to be a test file. If it was imported, it will be read. Did you mean to use `test.coverage.exclude`?

**Comment by jasonkuhrt** [comment]
Ok apparently `./xxx` and `xxx` lead to different results with regards to providing relative paths 🙈 ... This is quite frustrating, can't tell what is expected behaviour, versus bug, no docs about it, etc.

**Reply by AriPerkkio** [reply]
All these errors look like bug. My guess is that the root cause is in Remix's Vite plugin. Or maybe it is causing a weird edge case that we have never seen in other frameworks plugins.

**Comment by induleaf** [comment]
Is there any update regarding this issue? I'm also getting exactly the same issue after updating the Remix version 2, vitest coverage is failing.
![Screenshot 2024-01-22 at 10 12 13 AM](https://github.com/vitest-dev/vitest/assets/76951232/883f1ac0-2060-4571-92a4-b9dd4532b510)

**Reply by AriPerkkio** [reply]
Nope, no bug reports regarding this has been filed.

**Comment by weallblind** [comment]
I had the same issue and fixed it by set "sourceMap": true in tsconfig.json

**Reply by wangmeijian** [reply]
have not tsconfig.json，how to fix?

**Reply by induleaf** [reply]
Setting "sourceMap": true in tsconfig.json doesn't fix the issue for me. Is there some other config to be changed?

**Reply by weallblind** [reply]
That was pretty much it except I deleted compiled folder and recompiled

**Comment by RobertGemmaJr** [comment]
Also using Remix. I was able to find this by matching all instances of the build folder:

```typescript
export default defineConfig({
  // ...
  test: {
    // ...
    coverage: {
      exclude: [
        // ...
        "**/build/**",
      ],
    },
  },
});
```
